### ISSUE TITLE
Unknown attributes confuse debugger contributions silently

### ISSUE THREAD

**Issue by weinand** [microsoft/vscode#75887]
From Gitter (https://gitter.im/Microsoft/vscode?at=5d0a613526477710bbea174b):
A debugger author was contributing this debugger in his extension's package.json:

```
	"contributes": {
		"debuggers": [
			{
				"type": "papyrus",
				"label": "Papyrus Debug",
				"contributionAttributes": {
					"attach": {
						"properties": {
							"game": {
								"type": "string",
								"enum": [
									"fallout4"
								]
							},
							"projectPath": {
								"type": "string"
							}
						},
						"required": [
							"game"
						]
					}
				}
			}
		]
	},
```
But when running his extension the debug type "papyrus" was not defined.

After some investigation I noticed that "contributionAttributes" is the wrong name for the schema contribution. The correct name is "configurationAttributes".

I see two bugs here:
- why is the name of an incorrect attribute "contributionAttributes" not flagged as an error?
- why is the debug contribution "papyrus" ignored in this case?

**Comment by isidorn** [comment]
1. `configurationAttibutes` is not a required property - thus there is no error that it is missing. Extension authors contribute debuggers with no `configuraitonAttributes` and that is perfectly fine. It is possible to put any attribute value there, thus `contributionAttributes` is not flagged as an error. So this behavior is as defined.
2. Ignored papyrus when doing what exactly? We do not offer it to fill the launch.json since it does not contribute that. It shuold not be flaged as a warning when you have `type: papyrus` in `launch.json`

Thanks

**Comment by isidorn** [comment]
After discussing in standup we can improve 1 by saying that only known attributes are supported. I can look into changing that.

**Comment by weinand** [comment]
@isidorn with the typo in the package.json I get this error in a launch config that tries to use the debug type "papyrus":

![2019-06-21_15-07-00](https://user-images.githubusercontent.com/1898161/59924602-624eff80-9436-11e9-8229-a41fda9705b6.png)

**Comment by isidorn** [comment]
I have pushed a fix that we show a warning when a user types an unsported attribute inside contributes/debuggers or contributes/breakpoints.

As for that error the issue is that the papyrus debugger is not cotnributing the attach request. If you remove the attach part he will match it properly.
The actual issue is that the error message is not precise. Since he can not match the papyrus type with the configuratino that has a request attach. Unfortunetly the json schema does not give a fine grained control of the error messages, thus I always show that one. 
Closing since 1 is fixed and 2 is as designed
